app/controllers/items_controller.rb

```diff
@@ -19,16 +19,23 @@ def edit
     @item = Item.find(params[:id])
   end
 
-  def create
-    render_404 unless current_merchant?
-
+  def create    
     thumbnail = params[:item][:thumbnail]
-    File.open(Rails.root.join('app', 'assets', 'images', thumbnail.original_filename), 'wb') do |file|
-      file.write(thumbnail.read)
-    end
 
-    current_user.items.create(item_params)
-    redirect_to dashboard_items_path
+    if thumbnail
```

Comment:
This looks like code @stoic-plus has submitted. Definitely in items_controller.rb